server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/redirect/AddressExtractor.java

```diff
@@ -0,0 +1,146 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets.redirect;
+
+import java.util.List;
+import java.util.Locale;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+
+import org.apache.mailet.MailAddress;
+import org.apache.mailet.MailetContext;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Optional;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+
+public class AddressExtractor {
+
+    private final MailetContext mailetContext;
+
+    public AddressExtractor(MailetContext mailetContext) {
+        this.mailetContext = mailetContext;
+    }
+
+    public List<MailAddress> from(String addressList, List<String> allowedSpecials) throws MessagingException {
+        try {
+            return toMailAddresses(ImmutableList.copyOf(InternetAddress.parse(addressList, false)), allowedSpecials);
```

Comment:
you don't need to copy the list, you are the only one the have a reference on it. Or does it generate a raw array ?